jetty-server/src/main/java/org/eclipse/jetty/server/RejectInvalidAuthorityCustomizer.java

```diff
@@ -0,0 +1,77 @@
+//
+//  ========================================================================
+//  Copyright (c) 1995-2021 Mort Bay Consulting Pty Ltd and others.
+//  ------------------------------------------------------------------------
+//  All rights reserved. This program and the accompanying materials
+//  are made available under the terms of the Eclipse Public License v1.0
+//  and Apache License v2.0 which accompanies this distribution.
+//
+//      The Eclipse Public License is available at
+//      http://www.eclipse.org/legal/epl-v10.html
+//
+//      The Apache License v2.0 is available at
+//      http://www.opensource.org/licenses/apache2.0.php
+//
+//  You may elect to redistribute this code under either of these licenses.
+//  ========================================================================
+//
+
+package org.eclipse.jetty.server;
+
+import org.eclipse.jetty.http.BadMessageException;
+import org.eclipse.jetty.http.HttpStatus;
+import org.eclipse.jetty.http.HttpURI;
+import org.eclipse.jetty.util.StringUtil;
+
+/**
+ * Reject requests that have issues with their request level authority.
+ *
+ * <p>
+ * This addresses requests that either have schemes with no authority
+ * per https://datatracker.ietf.org/doc/html/rfc7230#section-5.4
+ * Or badly declared requests that have a non-absolute URI in the request line and an empty `Host` header.
+ * </p>
+ *
+ * <p>
+ * This will return a 400 Bad Request on failed authority checks.
+ * </p>
+ */
+public class RejectInvalidAuthorityCustomizer implements HttpConfiguration.Customizer
+{
+    @Override
+    public void customize(Connector connector, HttpConfiguration channelConfig, Request request)
+    {
+        if (!isValidAuthority(request.getHttpURI()))
```

Comment:
Is this correct? Don't we just care that there is an Authority in eitehr the URI or Host header (and if both then they match)?
Or has this been handled for us already and any Host header already copied over to the URI authority?

Reply:
Handled differently now, in `HostPort`.